### PR TITLE
Bump webpack from v4 to v5

### PR DESCRIPTION
**Pull request by ChristophWurst**
To review only look at the first commit. The rest is to update the branch and apply eslint fixes.

Formatting fixes split off into smaller PRs
* [x] https://github.com/nextcloud/mail/pull/6846
* [x] https://github.com/nextcloud/mail/pull/6847
* [x] https://github.com/nextcloud/mail/pull/6848

### CODE REVIEW

**Comment by ChristophWurst** [comment]
This includes a eslint config bump and there are _lots_ of formatting changes. Fun.

**Comment by kesselb** [comment]
> If you have rules defined for loading assets using raw-loader, url-loader, or file-loader, please use [Asset Modules](https://webpack.js.org/guides/asset-modules/) instead as they're going to be deprecated in near future.

That's something for a follow up I guess.

> Want to use environment variables with process.env.VARIABLE? You need to use the DefinePlugin or EnvironmentPlugin to define these variables in the configuration. Consider using VARIABLE instead and make sure to check typeof VARIABLE !== 'undefined' too. process.env is Node.js specific and should be avoided in frontend code.

We are using process.env in babel.config.js

**Comment by st3iny** [comment]
> > Want to use environment variables with process.env.VARIABLE? You need to use the DefinePlugin or EnvironmentPlugin to define these variables in the configuration. Consider using VARIABLE instead and make sure to check typeof VARIABLE !== 'undefined' too. process.env is Node.js specific and should be avoided in frontend code.
> 
> We are using process.env in babel.config.js

I think we are fine because `babel.config.js` is not shipped with our frontend. This warning is targeted at use cases which use env variables directly inside frontend code. I did a debug print inside `babel.config.js` and works as intended.

File: webpack.prod.js [hunk 1]
```diff
@@ -3,5 +3,5 @@ const common = require('./webpack.common.js')
 
 module.exports = merge(common, {
   mode: 'production',
-  devtool: '#source-map'
+  devtool: 'cheap-source-map'
```
[the remark this answers is not in the document]
Comment (st3iny) [reply webpack.prod.js]: I pushed a commit to address this.

**Review by st3iny — APPROVED** [review]
Tested and works.

**Comment by kesselb** [comment]
> Lint / ESLint (pull_request) 

Is failing. Our version of nextcloud/eslint-config is very old. 

https://github.com/nextcloud/eslint-config/releases/tag/v5.0.0 is with webpack 5.
https://github.com/nextcloud/eslint-config/releases the latest version is 8.0.0.

**Comment by st3iny** [comment]
Yeah, it turns out that we are still using npm@6 in some workflows (lint + test) because it is installed by default with Node 14. However, we required Node >= 14 and npm >= 7 in our `package.json`.

I'll open a new PR to fix this and then we can rebase and merge.

**Comment by st3iny** [comment]
I pulled the changes from #6911. A rebase was not possible due to conflicts so I did a merge.

**Pull request merged**